lib/api/navigation.js

```diff
@@ -0,0 +1,130 @@
+var linz = require('../../'),
+    utils = require('../utils'),
+    async = require('async');
+
+function get (req, cb) {
+
+    if (!linz.get('disable navigation cache') && !req.linz.cache.navigation.invalidate && req.session.navigation) {
+        return cb(null, req.session.navigation);
+    }
+
```

Comment:
Could we please have some debug output here saying `Priming navigation cache.` The debug should use the `debug` module, with a setting of `linz:cache`. That will allow us to get a handle of when the cache is/isn't being regenerated.